packages/snaps-rpc-methods/src/permitted/getCurrencyRate.ts

```diff
@@ -0,0 +1,108 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  Currency,
+  GetCurrencyRateParams,
+  GetCurrencyRateResult,
+  JsonRpcRequest,
+  Rate,
+} from '@metamask/snaps-sdk';
+import { type InferMatching } from '@metamask/snaps-utils';
+import {
+  StructError,
+  create,
+  literal,
+  object,
+  union,
+} from '@metamask/superstruct';
+import type { PendingJsonRpcResponse } from '@metamask/utils';
+
+import type { MethodHooksObject } from '../utils';
+
+const hookNames: MethodHooksObject<GetCurrencyRateMethodHooks> = {
+  getCurrencyRate: true,
+};
+
+export type GetCurrencyRateMethodHooks = {
+  /**
+   * @param currency - The currency symbol.
+   * Currently only 'btc' is supported.
```

Comment:
Can we somehow make this hook into the other rate controller?

Reply:
Yes it's the same return type :) The only thing is that rates get pulled only for the currently selected account in the other controller.

Reply:
The wallet API team may be working on changing that actually. We can keep as-is for now and sync with them.